src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -263,11 +263,10 @@ def readClusterSettings(self) -> None:
         self._tags = {k: v for k, v in (self.getLeader().tags or {}).items() if k != _TAG_KEY_TOIL_NODE_TYPE}
         # Grab the ARN name of the instance profile (a str) to apply to workers
         leader_info = None
-        for attempt in old_retry(timeout=300, predicate=lambda e: isinstance(e, ValueError)):
+        for attempt in old_retry(timeout=300, predicate=lambda e: True):
```

Comment:
We want to be careful with this too because this will retry on e.g. `ec2_metadata` not being defined, or other errors we would like to treat as genuine permanent problems. But the code here might be small enough that we can get away with it.

Reply:
It might be best to just define a new exception local to the scope of where ec2metadata is imported and being called.